src/main/java/sorald/processor/SoraldAbstractProcessor.java

```diff
@@ -39,10 +39,10 @@ public SoraldAbstractProcessor initResource(String originalFilesPath, File baseD
             if (file.isFile()) {
                 filesToScan.add(file.getAbsolutePath());
             } else {
-                try (Stream<Path> walk = Files.walk(Paths.get(file.getAbsolutePath()))) {
+                try {
                     filesToScan =
-                            walk.map(x -> x.toFile().getAbsolutePath())
-                                    .filter(f -> f.endsWith(Constants.JAVA_EXT))
```

Comment:
This is what caused the crash; it doesn't check if the file is actually a regular file (so it could be a directory ending with `.java`).